### ISSUE TITLE
Piece: Facebook Leads Trigger

### ISSUE THREAD

**Issue by enrico-bagattin** [activepieces/activepieces#933]
## Facebook Leads Trigger

In Facebook there are 4 entities to take into consideration: User Accounts, Facebook Pages, Forms, Leads

Pages are accessed through a FB account that has control of that page, a page could have one or more forms that retrieves leads.

### Use cases

Mainly there are three uses cases on Facebook Leads:
- retrieve leads per page (retrieving new leads for all forms of that page)
- retrieve leads per form of a specific page
- nice to have: It would be great also to have the possibility to select multiple forms of a page but not all of them

### Keep in mind

- many times there is the need to connect multiple accounts to access different pages (like other providers do) 
- forms are selectable only when Facebook ads campaigns are public, so the connectors must be retroactive to collect all the older leads
- Facebook forms could have custom fields

### My specific need

I am interested in this to connect FB to other platforms through webhooks or to save them in a Google Spreadsheet

**Comment by Lieb777** [comment]
Has there been any progress on this please?

**Comment by JanHolger** [comment]
it's currently in Todo status so there is no progress done yet but it's considered for development. If there is a need for it I would be willing to contribute here but I would need someone to help me understand the domain since I have no idea how FB marketing works and possibly also help with testing since I don't have any facebook page to test this with

**Comment by JanHolger** [comment]
blocked by #863

**Comment by MoShizzle** [comment]
I'll take this piece forward, will get started on it today and keep you updated

**Comment by ashrafsamhouri** [comment]
I assigned it to you @MoShizzle but as @JanHolger said it'd probably blocked by #863 which should be released tomorrow, so you'll have to check with the team
